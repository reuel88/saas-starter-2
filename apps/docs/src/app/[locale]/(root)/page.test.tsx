import { render, screen } from "@testing-library/react";
import Page from "./page";

jest.mock("next/navigation", () => {
  const original = jest.requireActual("next/navigation");

  return {
    ...original,
    useRouter() {
      return {
        push: jest.fn(),
      };
    },
  };
});

it("should render the main content with correct styles and layout", async () => {
  const { baseElement } = render(await Page({ params: { locale: "en" } }));

  expect(screen.getByRole("main")).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
