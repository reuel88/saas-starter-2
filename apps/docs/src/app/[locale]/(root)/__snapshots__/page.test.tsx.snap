// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the main content with correct styles and layout 1`] = `
<body>
  <div>
    <main
      class="flex min-h-screen flex-col items-center justify-between p-24"
    >
      <div
        class="z-10 flex w-full max-w-5xl items-center justify-between font-mono text-sm"
      >
        <p
          class="flex w-auto justify-center rounded-xl border border-neutral-800 bg-gradient-to-b from-zinc-200 p-4 backdrop-blur-2xl lg:bg-gray-200 dark:bg-zinc-800/30 dark:from-inherit lg:dark:bg-zinc-800/30"
        >
          examples/basic 
          <code
            class="font-mono font-bold"
          >
            docs
          </code>
        </p>
        <div
          class="flex items-end justify-center"
        >
          <a
            class="pointer-events-auto flex gap-2"
            href="https://vercel.com?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo"
            rel="noopener noreferrer"
            target="_blank"
          >
            By
             
            <img
              alt="Vercel Logo"
              class="dark:invert"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="24"
              src="/vercel.svg"
              style="color: transparent;"
              width="100"
            />
          </a>
        </div>
      </div>
      <div>
        <h1>
          My Internationalized App
        </h1>
        <div>
          <h3>
            Have a great day!
          </h3>
        </div>
        <a
          href="/about"
        >
          Go to Page 2
        </a>
        <div>
          <select
            class="text-black"
          >
            <option
              value="en-AU"
            >
              English
            </option>
            <option
              value="de"
            >
              Deutsch
            </option>
            <option
              value="ja"
            >
              日本語
            </option>
          </select>
        </div>
      </div>
      <div
        class="relative z-[-1] flex place-items-center before:absolute before:h-[300px] before:w-full before:-translate-x-1/2 before:rounded-full before:bg-gradient-radial before:from-white before:to-transparent before:blur-2xl before:content-[''] after:absolute after:-z-20 after:h-[180px] after:w-full after:translate-x-1/3 after:bg-gradient-conic after:from-sky-200 after:via-blue-200 after:blur-2xl after:content-[''] sm:before:w-[480px] sm:after:w-[240px] before:lg:h-[360px] before:dark:bg-gradient-to-br before:dark:from-transparent before:dark:to-blue-700 before:dark:opacity-10 after:dark:from-sky-900 after:dark:via-[#0141ff] after:dark:opacity-40"
      >
        <img
          alt="Next.js Logo"
          class="relative dark:drop-shadow-[0_0_0.3rem_#ffffff70] dark:invert"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="37"
          src="/next.svg"
          style="color: transparent;"
          width="180"
        />
      </div>
      <div
        class="mb-32 grid text-center lg:mb-0 lg:w-full lg:max-w-5xl lg:grid-cols-4 lg:text-left"
      >
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Docs
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Find in-depth information about Turborepo features and API.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs/handbook?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Learn
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Learn more about monorepos with our handbook.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs/getting-started/from-example?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Templates
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Choose from over 15 examples and deploy with a single click.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://vercel.com/new?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Deploy
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Instantly deploy your Turborepo to a shareable URL with Vercel.
          </p>
        </a>
      </div>
    </main>
  </div>
</body>
`;
